trio/tests/test_subprocess.py

```diff
@@ -570,3 +570,28 @@ async def test_for_leaking_fds():
     with pytest.raises(PermissionError):
         await run_process(["/dev/fd/0"])
     assert set(SyncPath("/dev/fd").iterdir()) == starting_fds
+
+
+async def test_subprocess_pidfd_unnotified():
+    noticed_exit = None
+
+    async def wait_and_tell(proc) -> None:
+        nonlocal noticed_exit
+        noticed_exit = False
+        await proc.wait()
+        noticed_exit = True
+
+    proc = await open_process(SLEEP(9999))
+    async with _core.open_nursery() as nursery:
+        nursery.start_soon(wait_and_tell, proc)
+        await wait_all_tasks_blocked()
+        assert noticed_exit is False
+        proc.terminate()
+        # without giving trio a chance to do so,
+        with assert_no_checkpoints():
+            # wait until the process has actually exited;
+            proc._proc.wait()
+            # force a call to poll (that closes the pidfd on linux)
+            proc.poll()
+        await wait_all_tasks_blocked()
```

Comment:
`wait_all_tasks_blocked()` isn't sufficient to guarantee that wait() should have completed on platforms where the waiting is done in another thread (like Windows -- I think that's why your tests are failing). I guess you could have `wait_and_tell` set an event, and wait on it with a timeout?

Reply:
Hmm, yeah. I was specifically hoping to avoid a timeout (my first attempts just used `fail_after` around the entire nursery).

I'll do that, though. I just always feel apprehensive about setting timeouts in test cases here.